cmd/infra/powervs/destroy.go

```diff
@@ -555,3 +622,75 @@ func deleteVpcSubnet(id string, v1 *vpcv1.VpcV1, infraID string) (err error) {
 	err = wait.PollImmediate(pollingInterval, vpcResourceDeletionTimeout, f)
 	return
 }
+
+// destroyVpcLB destroys VPC Load Balancer
+func destroyVpcLB(options *DestroyInfraOptions, subnetID string, v1 *vpcv1.VpcV1) error {
+
+	f := func(start string) (isDone bool, nextUrl string, err error) {
+
+		listLBOpt := vpcv1.ListLoadBalancersOptions{}
+		if start != "" {
+			listLBOpt.Start = &start
+		}
+		loadBalancerL, _, err := v1.ListLoadBalancers(&listLBOpt)
+		if err != nil {
+			return
+		}
+
+		if loadBalancerL == nil || len(loadBalancerL.LoadBalancers) <= 0 {
+			log(options.InfraID).Info("no load balancers available")
+			isDone = true
+			return
+		}
+
+		// Consider deleting LB which starting with 'kube-<cluster-name>'
+		// Which are provisioned by cloud controller manager. Below is the code ref for the naming convention.
+		// https://github.com/openshift/cloud-provider-powervs/blob/master/pkg/vpcctl/vpc_provider.go#L235
+		lbName := fmt.Sprintf("%s-%s", vpcLbNamePrefix, options.Name)
+
+		for _, lb := range loadBalancerL.LoadBalancers {
+			for _, subnet := range lb.Subnets {
+				if *subnet.ID == subnetID {
+					if strings.Contains(*lb.Name, lbName) {
+						err = deleteVpcLB(*lb.ID, v1, options.InfraID)
+						if err != nil {
+							return
+						}
+					}
+				}
+			}
+		}
+
+		// For paging over next set of resources getting the start token and passing it for next iteration
+		if loadBalancerL.Next != nil && *loadBalancerL.Next.Href != "" {
+			nextUrl = *loadBalancerL.Next.Href
+			return
+		}
+
+		isDone = true
+		return
+	}
+
+	return pagingHelper(f)
+}
+
+// deleteVpcLB deletes the load balancer id passed
+func deleteVpcLB(id string, v1 *vpcv1.VpcV1, infraID string) (err error) {
```

Comment:
I feel there will be confusion of `deleteVpcLB` vs `destroyVpcLB`, to keep it simple I'm wondering if we can move this code into `destroyVpcLB` as a simple inline function which takes id as arg